google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITFloat32Test.java

```diff
@@ -0,0 +1,448 @@
+            + ") PRIMARY KEY (Key)"
+      };
+
+  private static final String[] POSTGRESQL_SCHEMA =
+      new String[] {
+        "CREATE TABLE T ("
+            + "  Key                 VARCHAR PRIMARY KEY,"
+            + "  Float32Value        REAL,"
+            + "  Float32ArrayValue   REAL[]"
+            + ")"
+      };
+
+  private static DatabaseClient client;
+
+  private static boolean isUsingCloudDevel() {
+    String jobType = System.getenv("JOB_TYPE");
+
+    // Assumes that the jobType contains the string "cloud-devel" to signal that
+    // the environment is cloud-devel.
+    return !isNullOrEmpty(jobType) && jobType.contains("cloud-devel");
+  }
+
+  @BeforeClass
+  public static void setUpDatabase()
+      throws ExecutionException, InterruptedException, TimeoutException {
+    if (!isUsingCloudDevel()) {
+      return;
+    }
+
+    Database googleStandardSQLDatabase =
+        env.getTestHelper().createTestDatabase(GOOGLE_STANDARD_SQL_SCHEMA);
+
+    googleStandardSQLClient = env.getTestHelper().getDatabaseClient(googleStandardSQLDatabase);
+
+    Database postgreSQLDatabase =
+        env.getTestHelper()
+            .createTestDatabase(Dialect.POSTGRESQL, Arrays.asList(POSTGRESQL_SCHEMA));
+    postgreSQLClient = env.getTestHelper().getDatabaseClient(postgreSQLDatabase);
+  }
+
+  @Before
+  public void before() {
+    assumeFalse("Emulator does not support FLOAT32 yet", isUsingEmulator());
+
+    client =
+        dialect.dialect == Dialect.GOOGLE_STANDARD_SQL ? googleStandardSQLClient : postgreSQLClient;
+  }
+
+  @AfterClass
+  public static void tearDown() throws Exception {
+    ConnectionOptions.closeSpanner();
+  }
+
+  /** Sequence used to generate unique keys. */
+  private static int seq;
+
+  private static String uniqueString() {
+    return String.format("k%04d", seq++);
+  }
+
+  private String lastKey;
+
+  private Timestamp write(Mutation m) {
+    return client.write(Collections.singletonList(m));
+  }
+
+  private Mutation.WriteBuilder baseInsert() {
+    return Mutation.newInsertOrUpdateBuilder("T").set("Key").to(lastKey = uniqueString());
+  }
+
+  private Struct readRow(String table, String key, String... columns) {
+    return client
+        .singleUse(TimestampBound.strong())
+        .readRow(table, Key.of(key), Arrays.asList(columns));
+  }
+
+  private Struct readLastRow(String... columns) {
+    return readRow("T", lastKey, columns);
+  }
+
+  @Test
+  public void writeFloat32() {
+    assumeTrue("FLOAT32 is currently only supported in cloud-devel", isUsingCloudDevel());
+
+    write(baseInsert().set("Float32Value").to(2.0f).build());
+    Struct row = readLastRow("Float32Value");
+    assertThat(row.isNull(0)).isFalse();
+    assertThat(row.getFloat(0)).isWithin(0.0f).of(2.0f);
+  }
+
+  @Ignore("Needs a fix in the backend.")
+  @Test
+  public void writeFloat32NonNumbers() {
+    assumeTrue("FLOAT32 is currently only supported in cloud-devel", isUsingCloudDevel());
```

Comment:
This should not be needed, as the entire test class is conditional on that.

Reply:
Ack. I intended to write another test that verifies that FLOAT32 is not supported in production, but then realized that it would be a bad idea as the client tests would break when the backend starts supporting the type.